kopf/_cogs/clients/watching.py

```diff
@@ -79,8 +82,19 @@ async def infinite_watch(
                     namespace=namespace,
                     operator_pause_waiter=operator_pause_waiter,
                 )
-                async for raw_event in stream:
-                    yield raw_event
+                try:
+                    async for raw_event in stream:
+                        yield raw_event
+                except errors.APIClientError as ex:
+                    if ex.code != HTTP_TOO_MANY_REQUESTS_CODE:
+                        raise
+
+                    retry_wait = ex.details.get("retryAfterSeconds") or DEFAULT_RETRY_DELAY_SECONDS
```

Comment:
@wjsi quick question - is this key correct? Does `APIClientError` do some parsing/formatting of the response headers? When we saw this issue our response looked like:

```python
<ClientResponse(https://kubernetes.default.svc/apis/monitoring.coreos.com/v1/namespaces/tenant-sta-cwdev04-added01/servicemonitors/knative-proxy) [429 Too Many Requests]> <CIMultiDictProxy('Cache-Control': 'no-cache, private', 'Content-Type': 'text/plain; charset=utf-8', 'Retry-After': '1', 'X-Content-Type-Options': 'nosniff', 'Date': 'Tue, 14 Mar 2023 16:47:45 GMT', 'Content-Length': '43')>
```
So the response header is just `Retry-After` in our example.